src/main/java/com/farao_community/farao/dichotomy/Index.java

```diff
@@ -90,8 +87,9 @@ boolean precisionReached() {
         if (higherSecureStep != null && Math.abs(higherSecureStep.stepValue() - maxValue) < EPSILON) {
             return true;
         }
-        double minInterval = higherSecureStep != null ? higherSecureStep.stepValue() : minValue;
-        double maxInterval = lowerUnsecureStep != null ? lowerUnsecureStep.stepValue() : maxValue;
-        return Math.abs(minInterval - maxInterval) < precision;
+        if (lowerUnsecureStep == null || higherSecureStep == null) {
+            return false;
+        }
```

Comment:
There is `if (lowerUnsecureStep == null && higherSecureStep == null) {
            return false;
        }` just above in the code. I think you can remove it even if it can spare some operations. Or you can comment each case to make it more readable.

Reply:
Yep, done